### PR TITLE
BFB-33: Added Dash Mechanic

### PR DESCRIPTION
**Pull request by m4rtint**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-33

**Description**
- Added Separate Script to add Dash functionality
- Created a BFB Input action for Dash
- The Logic is mostly from the Sample - simple Dash, with a bit of Advanced Dash mixed in
- Architecture:

- Passed in base `Character` and a `Func<Vector2>` into the Dash Component. 
These are the main components Dash is based off of. `Character` will rarely change, so passing that in is best. 

**Screenshots / GIFs**

Screen Shot 2021-10-13 at 2 32 06 AM

Video: 
https://www.youtube.com/watch?v=L4rPZK5ROeA

### CODE REVIEW

File: Assets/_BForBoss/_Character/Scenes/Sandbox.unity [hunk 1]
```diff
@@ -3103,5 +3103,9 @@ PrefabInstance:
       propertyPath: _camera
       value: 
       objectReference: {fileID: 1280265869}
+    - target: {fileID: 7177967516030943659, guid: 9cef2f6c7c3da6449a278b7a673ecbd2, type: 3}
```
Comment (m4rtint) [inline Assets/_BForBoss/_Character/Scenes/Sandbox.unity:3106]: i can revert this if you want @unoctium1 , i created the custom input action for my dash.

Comment (m4rtint) [reply Assets/_BForBoss/_Character/Scenes/Sandbox.unity:3106]: It's entierly up to you. so i'll update the prefab

File: Assets/_BForBoss/_Character/Scripts/PlayerDashBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        {
+            _baseCharacter = baseCharacter;
+            _characterInputMovement = characterMovement;
+        }
+        
+        public void SetupPlayerInput(InputAction dashAction)
+        {
+            dashAction.started += OnDash;
+            dashAction.canceled += OnDash;
+            dashAction.Enable();
```
Comment (unoctium1) [inline Assets/_BForBoss/_Character/Scripts/PlayerDashBehaviour.cs:35]: I think you might want to be also disabling this in on disable, I'm not sure how important it is but I'm used to seeing that in input action code

**Comment by unoctium1** [comment]
Ah sorry typo, I meant cleaning up, in reference to the other comment about adding onDisable/onEnable for the input action

**Review by unoctium1 — APPROVED** [review]
Awesome, it looks great

**Pull request merged**